src/components/Dashboard/RewardsModule.js

```diff
@@ -15,46 +15,74 @@ import NoRewards from './NoRewards'
 import { useWallet } from '../../providers/Wallet'
 import { useCourtConfig } from '../../providers/CourtConfig'
 import { getProviderFromUseWalletId } from '../../ethereum-providers'
+import useJurorSubscriptionFees from '../../hooks/useJurorSubscriptionFees'
+
 import { bigNum, formatTokenAmount } from '../../lib/math-utils'
 import { addressesEqual } from '../../lib/web3-utils'
 
-const useTotalDisputesFees = (arbitrableFees, appealFees) => {
-  const totalArbitrableFees = arbitrableFees
-    ? arbitrableFees.reduce((acc, fee) => acc.add(fee.amount), bigNum(0))
-    : bigNum(0)
+const useTotalFeeRewards = (arbitrableFees, appealFees, subscriptionFees) => {
+  return useMemo(() => {
+    const totalArbitrableFees = arbitrableFees
+      ? arbitrableFees.reduce((acc, fee) => acc.add(fee.amount), bigNum(0))
+      : bigNum(0)
+
+    const totalAppealFees = appealFees
+      ? appealFees.reduce((acc, fee) => acc.add(fee.amount), bigNum(0))
+      : bigNum(0)
 
-  const totalAppealFees = appealFees
-    ? appealFees.reduce((acc, fee) => acc.add(fee.amount), bigNum(0))
-    : bigNum(0)
+    const totalSubscriptionFees = subscriptionFees
+      ? subscriptionFees.reduce((acc, fee) => acc.add(fee.amount), bigNum(0))
+      : bigNum(0)
 
-  return { totalArbitrableFees, totalAppealFees }
+    return { totalArbitrableFees, totalAppealFees, totalSubscriptionFees }
+  }, [appealFees, arbitrableFees, subscriptionFees])
 }
 
-// Ruling Fees => ANJ fees
-// Dispute fees + Appeal Fees => DAI fees
+// anjRewards => ANJ => Settled onSettleReward
+// feeRewards => DAI => settled onSettleRewards and onSettleAppealDeposit
+// subscriptions fees => DAI => claimed directly from the CourtSubscription contract
+// Once rewards are settled, we must withdraw them from the treasury (onWithdraw)
+// As opposed to fee rewards, subscription fees are directly sent to the juror's wallet after claimed
 const RewardsModule = React.memo(function RewardsModule({
   rewards,
   treasury,
   loading,
-  onWithdraw,
-  onSettleReward,
+  onClaimSubscriptionFees,
   onSettleAppealDeposit,
+  onSettleReward,
+  onWithdraw,
 }) {
   const wallet = useWallet()
   const { feeToken } = useCourtConfig()
 
-  const { totalArbitrableFees, totalAppealFees } = useTotalDisputesFees(
-    rewards?.arbitrableFees,
-    rewards?.appealFees
+  // Subscriptions are fetched directly from the subscriptions contract
+  const [subscriptionFees, setSubscriptionFees] = useJurorSubscriptionFees()
+  const { anjRewards, feeRewards } = rewards || {}
+
+  const {
+    totalArbitrableFees,
+    totalAppealFees,
+    totalSubscriptionFees,
+  } = useTotalFeeRewards(
+    feeRewards?.arbitrableFees,
+    feeRewards?.appealFees,
+    subscriptionFees
   )
 
+  // We'll get the total juror's balance held in the treasury
   // TODO: Handle possible multiple tokens (fee token can change)
```

Comment:
is this TODO still valid?

Reply:
Yes it is, this would be the case were we have multiple fee tokens